src/backend/dx11/src/lib.rs

```diff
@@ -3421,6 +3423,51 @@ pub struct ImageView {
     dsv_handle: Option<ComPtr<d3d11::ID3D11DepthStencilView>>,
     rodsv_handle: Option<ComPtr<d3d11::ID3D11DepthStencilView>>,
     uav_handle: Option<ComPtr<d3d11::ID3D11UnorderedAccessView>>,
+    borrowed: bool,
+}
+
+impl Clone for ImageView {
+    fn clone(&self) -> Self {
+        Self {
+            subresource: self.subresource,
+            format: self.format,
+            rtv_handle: self.rtv_handle.clone(),
+            srv_handle: self.srv_handle.clone(),
+            dsv_handle: self.dsv_handle.clone(),
+            rodsv_handle: self.rodsv_handle.clone(),
+            uav_handle: self.uav_handle.clone(),
+            borrowed: false
+        }
+    }
+}
+
+impl ImageView {
+    fn clone_borrowed(&self) -> Self {
+        unsafe {
+                Self {
+                subresource: self.subresource,
+                format: self.format,
+                rtv_handle: self.rtv_handle.as_ref().map(|p| ComPtr::from_raw(p.as_raw())),
+                srv_handle: self.srv_handle.as_ref().map(|p| ComPtr::from_raw(p.as_raw())),
+                dsv_handle: self.dsv_handle.as_ref().map(|p| ComPtr::from_raw(p.as_raw())),
+                rodsv_handle: self.rodsv_handle.as_ref().map(|p| ComPtr::from_raw(p.as_raw())),
+                uav_handle: self.uav_handle.as_ref().map(|p| ComPtr::from_raw(p.as_raw())),
+                borrowed: true
+            }
+        }
+    }
+}
+
+impl Drop for ImageView {
+    fn drop(&mut self) {
+        if self.borrowed {
+            mem::forget(self.rtv_handle.take());
```

Comment:
ok, it sounds very much like we should just got with raw pointers
If they are too annoying to deal with, we can define a wrapper `struct RawPtr<T>(*mut T)` or something like that and implement the necessary properties there

Reply:
Alright, sounds reasonable.